armory/data/datasets.py

```diff
@@ -269,6 +269,72 @@ def parse_split_index(split: str):
     return "+".join(output_tokens)
 
 
+def filter_by_index(dataset: "tf.data.Dataset", index: list, dataset_size: int):
```

Comment:
Very minor point - if dataset_size is expected to be an int, why is it cast to an int later? Should it be a Union of int and another type?

Reply:
It's primarily so that if the input is not an int, it will fail fast.